native/dpotf2.go

```diff
@@ -0,0 +1,66 @@
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+// Dpotrf computes the cholesky decomposition of the symmetric positive definite
+// matrix a. If ul == blas.Upper, then a is stored as an upper-triangular matrix,
+// and a = U^T U is stored in place into a. If ul == blas.Lower, then a = L L^T
+// is computed and stored in-place into a. This is the unblocked version of the
+// algorithm. Dpotrf returns 0 if the decomposition is successful, otherwise
```

Comment:
> On May 7, 2015, at 2:05 PM, Dan Kortschak <notifications@github.com> wrote:
> 
> In native/dpotf2.go <https://github.com/gonum/lapack/pull/11#discussion_r29895939>:
> 
> > @@ -0,0 +1,66 @@
> > +package native
> > +
> > +import (
> > +	"math"
> > +
> > +	"github.com/gonum/blas"
> > +	"github.com/gonum/blas/blas64"
> > +)
> > +
> > +// Dpotrf computes the cholesky decomposition of the symmetric positive definite
> > +// matrix a. If ul == blas.Upper, then a is stored as an upper-triangular matrix,
> > +// and a = U^T U is stored in place into a. If ul == blas.Lower, then a = L L^T
> > +// is computed and stored in-place into a. This is the unblocked version of the
> > +// algorithm. Dpotrf returns 0 if the decomposition is successful, otherwise
> Sorry to be difficult on this, but is the indormation we are discarding here likely to be useful to someone? Making it an error allows that to continue to be used if needed.

Personally, I don’t know for sure, but I don’t really see how. If A is positive definite, then all of the eigenvalues are > 0. A not being positive definite implies one of them is not. However, you can reorder the rows and such of a matrix without changing the eigenvalues (only the order). Thus, it seems to me that the info can’t give you the eigenvalue that’s negative, as there isn’t any real order to the eigenvaules, they just are. Possibly you could “augment” the diagonal of the matrix to make it positive definite, but again, it’s not clear to me why you’d change the first entry that failed. The value on the diagonal is already different from when it started. The integer is also weird in this case, because it’s one-indexed in normal lapack.

I did some quick google searching and couldn’t find an example where the returned integer was used. For example, https://stat.duke.edu/~cr173/Sta523_Fa14/blas_lapack.html <https://stat.duke.edu/~cr173/Sta523_Fa14/blas_lapack.html>, doesn’t even say that that’s the return.